app/knowledge_panels.py

```diff
@@ -1,167 +1,244 @@
-    if value is not None:
-        query["value_tag"] = value
-        description = f"{value} {hunger_game_filter}"
-    questions_url = "https://hunger.openfoodfacts.org/questions"
-    if query:
-        questions_url += f"?{urlencode(query)}"
-    t_description = hungergame()
-    description = f"{t_description} {description}"
-    html = f"<p><a href='{questions_url}'>{description}</a></p>\n"
-    return {
-        "hunger-game": {
-            "elements": [
+class KnowledgePanels:
+    def __init__(
+        self,
+        facet: str,
+        value: Union[str, None] = None,
+        sec_facet: Union[str, None] = None,
+        sec_value: Union[str, None] = None,
+        country: Union[str, None] = None,
+    ):
+        self.facet = facet
+        self.value = value
+        self.sec_facet = sec_facet
+        self.sec_value = sec_value
+        self.country = country
+
+    def hunger_game_kp(self):
+        query = {}
+        questions_url = "https://hunger.openfoodfacts.org/questions"
+        facets = {self.facet: self.value, self.sec_facet: self.sec_value}
+        filtered = {k: v for k, v in facets.items() if k is not None}
+        facets.clear()
+        facets.update(filtered)
+        urls = set()
+        description = ""
+        html = []
+        if self.country is not None:
+            query["country"] = f"en:{self.country}"
+        if "country" in facets.keys():
+            country_value = facets.get("country")
+            query["country"] = f"en:{country_value}"
+            description += country_value
+            facets.pop("country")
+        if "brand" in facets.keys() and len(facets) == 1:
+            brand_value = facets.get("brand")
+            if brand_value is not None:
+                query["brand"] = brand_value
+                description += f"brand {brand_value}"
+            else:
+                query["type"] = "brand"
+                description += "brand"
+        elif "brand" in facets.keys() and len(facets) > 1:
+            brand_value = facets.get("brand")
+            if brand_value is not None:
+                query["brand"] = brand_value
+            else:
+                description = "brand"
+                urls.add((f"{questions_url}?{urlencode(query)}&type=brand", description))
+            facets.pop("brand")
+            for k, v in facets.items():
+                query["type"] = k
+                description = k
+                if v is not None:
+                    query["value_tag"] = v
+                    description += f" {v}"
+                urls.add((f"{questions_url}?{urlencode(query)}", description))
+        elif len(facets) >= 1:
+            for k, v in facets.items():
+                query["type"] = k
+                description = k
+                if v is not None:
+                    query["value_tag"] = v
+                    description += f" {v}"
+                urls.add((f"{questions_url}?{urlencode(query)}", description))
```

Comment:
This should simply be rewritten:

```suggestion
        brand_description = ""
        if facets.get("brand") and len(facets) > 1:
            # we have a brand and other facets, let's use brand as a filter
            brand_value = facets.pop("brand")
            query["brand"] = brand_value
            brand_description = f" for brand {brand_value}"
        # generate an url for each remaining facets
        for k, v in facets.items():
            query["type"] = k
            facet_description = k
            if v is not None:
                query["value_tag"] = v
                description += f" {v}"
            urls.add((
                f"{questions_url}?{urlencode(query)}", 
                f"questions about {facet_description}{brand_description}{description}"
            ))
```

See how simple it is :-)

Reply:
I changed it a bit, because we have to copy query before adding type and value.
If not, if you have first facet with a value and second without a value, the first facet value would have been used for the second facet.

Reply:
Thank you!